cms/core/tests/test_permissions.py

```diff
@@ -0,0 +1,165 @@
+
+    def test_publishing_admin_can_manage_glossary_term(self):
+        self.all_permissions_check_helper("core", "glossaryterm")
+
+    def test_publishing_admin_can_manage_contact_details(self):
+        self.all_permissions_check_helper("core", "contactdetails")
+
+    def test_publishing_admin_can_manage_main_menu(self):
+        self.all_permissions_check_helper("navigation", "mainmenu")
+        # also check that the PA can publish the main menu
+        self.assertTrue(self.user.has_perm("navigation.publish_mainmenu"))
+
+    def test_publishing_admin_can_manage_footer_menu(self):
+        self.all_permissions_check_helper("navigation", "footermenu")
+        # also check that the PA can publish the footer menu
+        self.assertTrue(self.user.has_perm("navigation.publish_footermenu"))
+
+    def test_publishing_admin_can_manage_bundles(self):
+        self.all_permissions_check_helper("bundles", "bundle")
+        # also check that the PA can view bundles
+        self.assertTrue(self.user.has_perm("bundles.view_bundle"))
+
+    def test_publishing_admin_can_manage_redirects(self):
+        self.all_permissions_check_helper("wagtailredirects", "redirect")
+
+    def test_publishing_admin_can_manage_navigation_settings(self):
+        self.assertTrue(self.user.has_perm("navigation.change_navigationsettings"))
+
+    def test_publishing_admin_can_manage_social_media_settings(self):
+        self.assertTrue(self.user.has_perm("core.change_socialmediasettings"))
+
+
+class PublishingOfficerPermissionsTestCase(PermissionsTestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.create_user(group_name=settings.PUBLISHING_OFFICERS_GROUP_NAME)
+
+    def test_publishing_officer_can_access_admin(self):
+        """Check that the Publishing Admin can access the Wagtail admin."""
+        self.assertTrue(self.user.has_perm("wagtailadmin.access_admin"))
+
+    def test_publishing_officer_can_manage_bundle(self):
+        """Check that the Publishing Officer can manage bundles."""
+        self.all_permissions_check_helper("bundles", "bundle")
+        # also check that the PO can view bundles
+        self.assertTrue(self.user.has_perm("bundles.view_bundle"))
+
+    def test_publishing_officer_can_manage_pages(self):
+        """Check that the Publishing Officer can create and edit the root page and its children."""
+        for permission_type in ("add", "change"):
+            self.page_permission_check_helper(permission_type)
+
+    def test_publishing_officer_can_choose_images(self):
+        """Check that the Publishing Officer can choose images on pages."""
+        self.collection_permission_check_helper("choose", "image")
+
+    def test_publishing_officer_can_choose_documents(self):
+        """Check that the Publishing Officer can choose documents on pages."""
+        self.collection_permission_check_helper("choose", "document")
+
+
+class ViewerPermissionsTestCase(PermissionsTestCase):
```

Comment:
Testing for Snippet permissions would be nice too.

Currently, I think you're assigning snippet permissions `add`, `change`, `delete` and `publish` to `Publishing Admins`?

So it would be good to confirm that `Publishing Officers` and `Viewers` lack snippet permissions.

Some negative tests around this will be awesome.

Reply:
https://github.com/ONSdigital/dis-wagtail/pull/128#discussion_r2026761530